### PR TITLE
Handle response from server better

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: go/client/logclient.go [hunk 1]
```diff
@@ -139,37 +148,77 @@ func (c *LogClient) fetchAndParse(uri string, res interface{}) error {
...
+		} else {
+			switch {
+			case httpResp.StatusCode == 200:
+				done = true
+				break
+			case httpResp.StatusCode == 408:
+			case httpResp.StatusCode == 503:
+				// Retry
+				backoffSeconds = 10
+				if retryAfter := httpResp.Header.Get("Retry-After"); retryAfter != "" {
```
Comment (pphaneuf) [inline go/client/logclient.go]: Whoa.

File: go/preload/main/preload.go [hunk 2]
```diff
@@ -70,7 +70,7 @@ func sctWriterJob(addedCerts <-chan *preload.AddedCert, sctWriter io.Writer, wg
 func certSubmitterJob(addedCerts chan<- *preload.AddedCert, log_client *client.LogClient, certs <-chan *client.LogEntry,
 	wg *sync.WaitGroup) {
 	for c := range certs {
-		chain := make([]client.ASN1Cert, len(c.Chain) + 1)
+		chain := make([]client.ASN1Cert, len(c.Chain)+1)
```
Comment (pphaneuf) [inline go/preload/main/preload.go]: gofmt wasn't run previously?

**Comment by pphaneuf** [comment]
LGTM

File: go/preload/main/preload.go (hunk 2 above)
Comment (AlCutter) [inline go/preload/main/preload.go]: Looks like that slipped through.

**Pull request merged**